### PR TITLE
Generalise the API slightly

### PR DESCRIPTION
**Pull request by adinapoli**
This small PR generalises a bit the API by introducing a new function called `sendMailSESWithResponse` which is _really_ what `sendMailSES` used to be, but it exposes a `onResponseStatus` callback  that can be used to customise the return type, to account for situations where users might want more control over what to do on the `Status` returned by the HTTP `Response`, and return something more meaningful than `()`.

`sendMailSES` is now re-implemented under the hood in terms of this new function, thus this is not a breaking change as the type signature for `sendMailSES` stays the same.

@kindaro What do you think?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 *.swp
 dist
 .stack-work/
+tags
+dist-newstyle
```
Comment (kindaro) [inline .gitignore:5]: This should go to a pull request of its own, together with `cabal.project`.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
 *.swp
 dist
 .stack-work/
+tags
```
Comment (kindaro) [inline .gitignore:4]: This line is best omitted from this pull request. Rather, we can craft a new pull request with all sorts of editor specific stuff, such as emacs and vim tags.

File: cabal.project [hunk 3]
```diff
@@ -0,0 +1,2 @@
+packages: ./mime-mail
+          ./mime-mail-ses
```
Comment (kindaro) [inline cabal.project:2]: This should go to a pull request of its own, dedicated to adding support for building with Cabal.

File: mime-mail-ses/Network/Mail/Mime/SES.hs [hunk 4]
```diff
@@ -58,14 +59,28 @@ renderSendMailSESGlobal ses mail = do
   mgr <- liftIO getGlobalManager
   renderSendMailSES mgr ses mail
 
-sendMailSES :: MonadIO m => Manager -> SES 
+
+sendMailSES :: MonadIO m
+            => Manager
+            -> SES
```
Comment (kindaro) [inline mime-mail-ses/Network/Mail/Mime/SES.hs:65]: I understand your desire to fix this inconsistent white space, however, being independent of all other changes, this one should go in its own _«white space»_ commit.

File: mime-mail-ses/Network/Mail/Mime/SES.hs [hunk 5]
```diff
@@ -58,14 +59,28 @@ renderSendMailSESGlobal ses mail = do
...
                             -- email header fields, MIME types, and
                             -- MIME encoding.
             -> m ()
-sendMailSES manager ses msg = liftIO $ do
+sendMailSES manager ses msg =
+  sendMailSESWithResponse manager ses msg checkForError
+
+-- | @since 0.4.3
+-- Generalised version of 'sendMailSES' which allows customising the final return type.
+sendMailSESWithResponse :: MonadIO m => Manager
```
Comment (kindaro) [inline mime-mail-ses/Network/Mail/Mime/SES.hs:77]: This here is asking for a line break before `=>` for consistency with the other white space change above.

**Review by kindaro — CHANGES_REQUESTED** [review]
## This is a reasonable addition and I think it should go in.

There are several independent changes here:
* White space fix — should go in its own commit or pull request.
* Cabal support — should go in its own pull request.
* Editor support _(tags)_ — should go in their own pull request, and maybe Emacs tags can be added too.
* The feature itself — this looks good.

This is of course all minutiae, but still I would appreciate if you can address it. We must make every effort to keep the history clear and complete.

**Comment by adinapoli** [comment]
Hey Ignat,

Thanks for getting back to me so quickly. Yes, all good; different projects
have different conventions when it comes to adding more (possibly
unrelated) stuff to a PR, so I went for the "beefy" approach by default, as
the rest is harmless, but I understand where you're coming from :)

I will polish and rebase the PR tomorrow so that it delivers only the API
change, and will fire other PRs with the stylistic changes as I see fit.

Thank you!

**Comment by adinapoli** [comment]
@kindaro Ok, I think we should be in business now, let me know if this is more palatable now!

**Review by kindaro — APPROVED** [review]
Nice! I shall make a Hackage release within a day.

**Comment by kindaro** [comment]
@adinapoli  I released to Hackage, hope everything matches your expectations. Let me know if anything is out of order.

**Comment by adinapoli** [comment]
Marvellous, thank you very much @kindaro !

I will try this out next week and report back, but from a quick Hackage inspection I think everything is fine 😉

**Pull request merged**